### PR TITLE
[pylint] checking that list_ method return a Paging class

### PR DESCRIPTION
**Pull request by l0lawrence**
Just wanted some review for the changing of inferring return type to inferring directly on the return node if it is a call to a function.

I have a TODO to flag methods that return ItemPaged, but are not named with list (i.e Search). 

No rush on this - not to be merged until after repo shift

### CODE REVIEW

File: scripts/pylint_custom_plugin/pylint_guidelines_checker.py [hunk 1]
```diff
@@ -886,17 +887,30 @@ def visit_functiondef(self, node):
...
-                        logger.debug("Pylint custom checker failed to check if client list method uses core paging.")
-                        pass
+                    paging_class = False
+
+                    for inner_node in node.body:
+                        # If this is a return node
+                        if isinstance(inner_node, astroid.Return):
+                            try:
+                                # Checking that what we are returning is a function call
+                                if isinstance(inner_node.value, astroid.Call):
```
Comment (kristapratico) [inline scripts/pylint_custom_plugin/pylint_guidelines_checker.py:902]: does this work for either an ItemPaged or a function call which returns an ItemPaged? If the return type is very incorrect, like returning `None` or something, will that get caught?

File: scripts/pylint_custom_plugin/tests/test_files/list_return_type_custom_class_acceptable.py [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+        query = None
+        return SearchItemPaged(
+            self._client, query, kwargs, page_iterator_class=None
+        )
+    
+    def list_this(self):
+        return azure.core.paging.ItemPaged()
+    
+    def list_something(self):
+        return [1,2,3,4,5]
```
Comment (kristapratico) [inline scripts/pylint_custom_plugin/tests/test_files/list_return_type_custom_class_acceptable.py:34]: files says acceptable, but shouldn't the linter error on this?

File: scripts/pylint_custom_plugin/tests/test_pylint_custom_plugins.py [hunk 3]
```diff
@@ -1886,27 +1921,42 @@ def test_ignores_methods_return_ItemPaged(self):
...
             async def list_thing(self): #@
+                '''
+                :rtype: AsyncItemPaged()
+                '''
                 return AsyncItemPaged()
+
             @distributed_trace
             def list_thing2(self): #@
+                '''
+                :rtype: AsyncItemPaged()
```
Comment (kristapratico) [inline scripts/pylint_custom_plugin/tests/test_pylint_custom_plugins.py:1958]: maybe do one with the full path? ~azure.core.async_paging.AsyncItemPaged

File: scripts/pylint_custom_plugin/tests/test_files/list_return_type_custom_class_acceptable.py (hunk 2 above)
Comment (l0lawrence) [reply scripts/pylint_custom_plugin/tests/test_files/list_return_type_custom_class_acceptable.py:34]: I updated it to a file of both errors and non-errors, I will update the title!

File: scripts/pylint_custom_plugin/pylint_guidelines_checker.py (hunk 1 above)
Comment (l0lawrence) [reply scripts/pylint_custom_plugin/pylint_guidelines_checker.py:902]: Good point, I will test with those!

**Pull request closed without merging**